diff --git a/PKGBUILD b/PKGBUILD
index 9ef1363..91bc68d 100644
--- a/PKGBUILD
+++ b/PKGBUILD
@@ -16,6 +16,7 @@ sha256sums=('9473f5454720eaf4713cb9937b8b249f5e8497a1e9c04bb241936ae1b07c07ca')
 build() {
   cd "$pkgname-$pkgver"
 
+  CFLAGS="$(echo "$CFLAGS" | sed -e s/mtune/march/ -e s/O2/Ofast/)"
   CFLAGS+=' -Wa,--noexecstack'
   ./autogen.sh
   ./configure --prefix=/usr --with-crypto --with-curl
