diff --git a/.SRCINFO b/.SRCINFO
index fb34076..36debe1 100644
--- a/.SRCINFO
+++ b/.SRCINFO
@@ -1,6 +1,6 @@
 pkgbase = curl-http3
 	pkgdesc = An URL retrieval utility and library - compiled with HTTP/3 support - binary is called curl3
-	pkgver = 7.73.0
+	pkgver = 7.74.0
 	pkgrel = 1
 	url = https://curl.haxx.se
 	arch = x86_64
@@ -16,10 +16,10 @@ pkgbase = curl-http3
 	depends = libnghttp2
 	depends = libidn2
 	depends = libidn2.so
-	source = https://curl.haxx.se/download/curl-7.73.0.tar.gz
-	source = https://curl.haxx.se/download/curl-7.73.0.tar.gz.asc
+	source = https://curl.haxx.se/download/curl-7.74.0.tar.gz
+	source = https://curl.haxx.se/download/curl-7.74.0.tar.gz.asc
 	validpgpkeys = 27EDEAF22F3ABCEB50DB9A125CC908FDB71E12C2
-	sha512sums = 86996f7ce537cfc39be18a6a93af0524c7a878d9e003ce0106eb4ff6ce2dea99da1cb9af297b87f93b204b42a3d793198a6e01764aa2a06f5b6fe0623b008f88
+	sha512sums = 4b61a23463315dce5529aa5f1dc7d21d7876347912c68855c9cfcb01e06492af1510975fafb213c67ac7b9764287767da69043a79376a4be366aa23ace09f163
 	sha512sums = SKIP
 
 pkgname = curl-http3
diff --git a/PKGBUILD b/PKGBUILD
index 1a5b7d8..c8ea502 100644
--- a/PKGBUILD
+++ b/PKGBUILD
@@ -1,6 +1,6 @@
 # Maintainer: Hendrik 'T4cC0re' Meyer <aur@t4cc0.re>
 pkgname=curl-http3
-pkgver=7.73.0
+pkgver=7.74.0
 pkgrel=1
 pkgdesc="An URL retrieval utility and library - compiled with HTTP/3 support - binary is called curl3"
 arch=('x86_64')
@@ -13,8 +13,8 @@ source=("https://curl.haxx.se/download/curl-$pkgver.tar.gz"{,.asc})
 validpgpkeys=('27EDEAF22F3ABCEB50DB9A125CC908FDB71E12C2') # Daniel Stenberg
 #source=("curl-$pkgver.zip::https://github.com/curl/curl/archive/master.zip")
 
-# Master as of 2020-10-14 (7.73.0 release day)
-_quiche_ref=107a054b73bb4d381931dbc46aecccbb1801d91a
+# Master as of 2020-12-10
+_quiche_ref=e3ce6eb5bd2347d6b46d69ce9786a6132028e140
 
 build(){
   rm -rf quiche
@@ -76,4 +76,4 @@ package() {
   # libquice is requried
   install -Dm755 quiche/target/release/libquiche.so "${pkgdir}/usr/lib/libquiche.so"
 }
-sha512sums=(86996f7ce537cfc39be18a6a93af0524c7a878d9e003ce0106eb4ff6ce2dea99da1cb9af297b87f93b204b42a3d793198a6e01764aa2a06f5b6fe0623b008f88             SKIP)
+sha512sums=(4b61a23463315dce5529aa5f1dc7d21d7876347912c68855c9cfcb01e06492af1510975fafb213c67ac7b9764287767da69043a79376a4be366aa23ace09f163             SKIP)
