diff --git a/.SRCINFO b/.SRCINFO
index 058dff0..e248b47 100644
--- a/.SRCINFO
+++ b/.SRCINFO
@@ -1,7 +1,7 @@
 pkgbase = kotatogram-desktop
 	pkgdesc = Kotatogram – experimental Telegram Desktop fork
-	pkgver = 1.3.4
-	pkgrel = 3
+	pkgver = 1.3.9
+	pkgrel = 1
 	url = https://kotatogram.github.io
 	arch = x86_64
 	license = GPL3
@@ -11,6 +11,8 @@ pkgbase = kotatogram-desktop
 	makedepends = ninja
 	makedepends = tl-expected
 	makedepends = range-v3
+	makedepends = microsoft-gsl
+	makedepends = libwebrtc
 	depends = qt5-base
 	depends = qt5-imageformats
 	depends = ffmpeg
@@ -24,15 +26,17 @@ pkgbase = kotatogram-desktop
 	depends = libdbusmenu-qt5
 	depends = hunspell
 	depends = hicolor-icon-theme
+	depends = qt5-wayland
+	depends = gtk3
 	optdepends = ttf-opensans: default Open Sans font family
 	conflicts = kotatogram-desktop-bin
 	conflicts = kotatogram-desktop-dynamic-bin
-	source = https://github.com/kotatogram/kotatogram-desktop/releases/download/k1.3.4/kotatogram-desktop-1.3.4-full.tar.gz
-	source = telegram-desktop-qt5.15.patch
-	source = 0001-Add-an-option-to-hide-messages-from-blocked-users-in.patch
-	sha512sums = f0255e1bcbf1703c684e3c71392156002456a988f4128107e0181b4f4b9956073c29c27ff19d8416b19b3d836da999b39a421bd9755683013eec71ff51104f38
-	sha512sums = 6bbe0583103d575a1413141799985c7c7b590d8a9856c727320ce322679a3d38343a144c314eee54fa9776c8f3a084627a333e6860d1b8cb8f496b1702070ac0
-	sha512sums = add35207d3a2772b3886f63d81751551e7ad5b6dac8ca3be4dc7bec6f126f04deaa4013d01805b524fbcdcfe7cacd94fa92c4583a62e7bd78472801d9beeb591
+	source = https://github.com/kotatogram/kotatogram-desktop/releases/download/k1.3.9/kotatogram-desktop-1.3.9-full.tar.gz
+	source = 0001-Update-webrtc-packaged-build-for-tg_owt.patch::https://github.com/kotatogram/cmake_helpers/commit/d955882cb4d4c94f61a9b1df62b7f93d3c5bff7d.patch
+	source = 0002-Use-tg_owt-webrtc-fork.patch::https://github.com/kotatogram/cmake_helpers/commit/4c8956027de8e8e8b984c5daa643aacb14a89123.patch
+	sha512sums = a63a8edd0fa56f717cce31005b41284af73fe87a16307aa6aa688664d6d3aa381a26d459bf3fcdd931d552e08756d187f54ed8c87425c44dee118402a4b5d4ad
+	sha512sums = b3c44e76a3907f7acc197746b471564577e912bf0561e9576dc8459211c88f400716437bcaa10967376461c69c8a98a56477d26d3feb9ca34747d9208bf5f6c6
+	sha512sums = 3a7a20ace7dcea5be202e7b3008807ba0ae38ae8266709e3595f6b3c0aa2bf81299da4b5a7e7d66d2d9b3040cc6618e67e4d7b9572b2649db4c477056e649bcf
 
 pkgname = kotatogram-desktop
 
diff --git a/PKGBUILD b/PKGBUILD
index 373c809..9f0328d 100644
--- a/PKGBUILD
+++ b/PKGBUILD
@@ -2,8 +2,8 @@
 # Contributor: Auteiy <dmitry@auteiy.me>
 
 pkgname=kotatogram-desktop
-pkgver=1.3.4
-pkgrel=3
+pkgver=1.3.9
+pkgrel=1
 pkgdesc="Kotatogram – experimental Telegram Desktop fork"
 arch=(x86_64)
 url="https://kotatogram.github.io"
@@ -22,6 +22,8 @@ depends=(
 	libdbusmenu-qt5
 	hunspell
 	hicolor-icon-theme
+	qt5-wayland
+	gtk3
 )
 makedepends=(
 	git
@@ -30,6 +32,8 @@ makedepends=(
 	ninja
 	tl-expected
 	range-v3
+	microsoft-gsl
+	libwebrtc
 )
 optdepends=(
 	'ttf-opensans: default Open Sans font family'
@@ -37,17 +41,18 @@ optdepends=(
 conflicts=('kotatogram-desktop-bin' 'kotatogram-desktop-dynamic-bin')
 source=(
 	"https://github.com/kotatogram/${pkgname}/releases/download/k${pkgver}/${pkgname}-${pkgver}-full.tar.gz"
-	"telegram-desktop-qt5.15.patch"
-	"0001-Add-an-option-to-hide-messages-from-blocked-users-in.patch"
+	"0001-Update-webrtc-packaged-build-for-tg_owt.patch::https://github.com/kotatogram/cmake_helpers/commit/d955882cb4d4c94f61a9b1df62b7f93d3c5bff7d.patch"
+	"0002-Use-tg_owt-webrtc-fork.patch::https://github.com/kotatogram/cmake_helpers/commit/4c8956027de8e8e8b984c5daa643aacb14a89123.patch"
 )
-sha512sums=('f0255e1bcbf1703c684e3c71392156002456a988f4128107e0181b4f4b9956073c29c27ff19d8416b19b3d836da999b39a421bd9755683013eec71ff51104f38'
-            '6bbe0583103d575a1413141799985c7c7b590d8a9856c727320ce322679a3d38343a144c314eee54fa9776c8f3a084627a333e6860d1b8cb8f496b1702070ac0'
-            'add35207d3a2772b3886f63d81751551e7ad5b6dac8ca3be4dc7bec6f126f04deaa4013d01805b524fbcdcfe7cacd94fa92c4583a62e7bd78472801d9beeb591')
+sha512sums=('a63a8edd0fa56f717cce31005b41284af73fe87a16307aa6aa688664d6d3aa381a26d459bf3fcdd931d552e08756d187f54ed8c87425c44dee118402a4b5d4ad'
+            'b3c44e76a3907f7acc197746b471564577e912bf0561e9576dc8459211c88f400716437bcaa10967376461c69c8a98a56477d26d3feb9ca34747d9208bf5f6c6'
+            '3a7a20ace7dcea5be202e7b3008807ba0ae38ae8266709e3595f6b3c0aa2bf81299da4b5a7e7d66d2d9b3040cc6618e67e4d7b9572b2649db4c477056e649bcf')
 
 prepare() {
-    cd ${pkgname}-$pkgver-full
-    patch -p1 < ${srcdir}/telegram-desktop-qt5.15.patch
-    patch -p1 < ${srcdir}/0001-Add-an-option-to-hide-messages-from-blocked-users-in.patch
+    cd ${pkgname}-$pkgver-full/cmake
+    patch -R -Np1 < ../../0001-Update-webrtc-packaged-build-for-tg_owt.patch
+    patch -R -Np1 < ../../0002-Use-tg_owt-webrtc-fork.patch
+    sed 's|set(webrtc_build_loc ${webrtc_loc}/out/$<CONFIG>/obj)|set(webrtc_build_loc /usr/lib)|' -i external/webrtc/CMakeLists.txt
 }
 
 build() {
@@ -57,10 +62,7 @@ build() {
 		-DCMAKE_INSTALL_PREFIX=/usr \
 		-DCMAKE_BUILD_TYPE=Release \
 		-DTDESKTOP_API_TEST=ON \
-		-DDESKTOP_APP_USE_PACKAGED_GSL=OFF \
-		-DDESKTOP_APP_USE_PACKAGED_RLOTTIE=OFF \
-		-DDESKTOP_APP_USE_PACKAGED_VARIANT=OFF \
-		-DTDESKTOP_USE_PACKAGED_TGVOIP=OFF
+		-DDESKTOP_APP_WEBRTC_LOCATION=/usr/include/libwebrtc
 
 	cmake --build build
 }
