diff --git a/.SRCINFO b/.SRCINFO
index d1bbdc5..a85c377 100644
--- a/.SRCINFO
+++ b/.SRCINFO
@@ -1,6 +1,6 @@
 pkgbase = lib32-libbluray
 	pkgdesc = Library to access Blu-Ray disks for video playback (32 bit)
-	pkgver = 1.2.0
+	pkgver = 1.2.1
 	pkgrel = 1
 	url = http://www.videolan.org/developers/libbluray.html
 	arch = x86_64
@@ -12,8 +12,8 @@ pkgbase = lib32-libbluray
 	depends = lib32-freetype2
 	depends = lib32-libxml2
 	provides = libbluray.so
-	source = ftp://ftp.videolan.org/pub/videolan/libbluray/1.2.0/libbluray-1.2.0.tar.bz2
-	sha512sums = d10413b6b86ff2d2e7c4b0103546f2142727cc5209ddb7b227aa74e27384f2e0b9abee37bf8ccc5b0cdfcaeebfb0669cf20903a247df278a8ad6dbd27469d324
+	source = ftp://ftp.videolan.org/pub/videolan/libbluray/1.2.1/libbluray-1.2.1.tar.bz2
+	sha512sums = e9eeee433034266e88c05ae3469c0bd2bf700d7f16e17ad706f48426c143c823e83c16ab79a637f1e85e4eb858aa0577630671f87ba35d179db2a1d8bc8545c4
 
 pkgname = lib32-libbluray
 
diff --git a/PKGBUILD b/PKGBUILD
index 5e30d76..6104ed9 100644
--- a/PKGBUILD
+++ b/PKGBUILD
@@ -5,7 +5,7 @@
 
 _pkgbase=libbluray
 pkgname=lib32-${_pkgbase}
-pkgver=1.2.0
+pkgver=1.2.1
 pkgrel=1
 pkgdesc="Library to access Blu-Ray disks for video playback (32 bit)"
 arch=('x86_64')
@@ -15,7 +15,7 @@ depends=("$_pkgbase" 'lib32-fontconfig' 'lib32-glibc' 'lib32-freetype2' 'lib32-l
 makedepends=('gcc-multilib')
 provides=('libbluray.so')
 source=("ftp://ftp.videolan.org/pub/videolan/$_pkgbase/$pkgver/$_pkgbase-$pkgver.tar.bz2")
-sha512sums=('d10413b6b86ff2d2e7c4b0103546f2142727cc5209ddb7b227aa74e27384f2e0b9abee37bf8ccc5b0cdfcaeebfb0669cf20903a247df278a8ad6dbd27469d324')
+sha512sums=('e9eeee433034266e88c05ae3469c0bd2bf700d7f16e17ad706f48426c143c823e83c16ab79a637f1e85e4eb858aa0577630671f87ba35d179db2a1d8bc8545c4')
 
 build() {
   export CC='gcc -m32'
