diff --git a/.SRCINFO b/.SRCINFO
index 64d98eb..f59ed3e 100644
--- a/.SRCINFO
+++ b/.SRCINFO
@@ -1,6 +1,6 @@
 pkgbase = lib32-srt
 	pkgdesc = Secure Reliable Transport library (32-bit)
-	pkgver = 1.4.1
+	pkgver = 1.4.2
 	pkgrel = 1
 	url = https://www.srtalliance.org/
 	arch = x86_64
@@ -11,10 +11,8 @@ pkgbase = lib32-srt
 	depends = lib32-gcc-libs
 	depends = lib32-openssl
 	depends = srt
-	source = git+https://github.com/Haivision/srt#commit=9e52606fae3b32818b5b734d45d3358f4a95cf3f
-	source = no-rpath.diff
+	source = git+https://github.com/Haivision/srt#commit=50b7af06f3a0a456c172b4cb3aceafa8a5cc0036
 	sha256sums = SKIP
-	sha256sums = ce9a97d661d796da85a7e4faf9c70c27368d00eb505f85c03a31bba8f22e6847
 
 pkgname = lib32-srt
 
diff --git a/PKGBUILD b/PKGBUILD
index 634cb1f..c9b54c7 100644
--- a/PKGBUILD
+++ b/PKGBUILD
@@ -2,7 +2,7 @@
 
 _basename=srt
 pkgname=lib32-srt
-pkgver=1.4.1
+pkgver=1.4.2
 pkgrel=1
 pkgdesc="Secure Reliable Transport library (32-bit)"
 url="https://www.srtalliance.org/"
@@ -10,11 +10,9 @@ arch=(x86_64)
 license=(MPL2)
 depends=(lib32-gcc-libs lib32-openssl srt)
 makedepends=(git cmake ninja)
-_commit=9e52606fae3b32818b5b734d45d3358f4a95cf3f  # tags/v1.4.1
-source=("git+https://github.com/Haivision/srt#commit=$_commit"
-        no-rpath.diff)
-sha256sums=('SKIP'
-            'ce9a97d661d796da85a7e4faf9c70c27368d00eb505f85c03a31bba8f22e6847')
+_commit=50b7af06f3a0a456c172b4cb3aceafa8a5cc0036  # tags/v1.4.2
+source=("git+https://github.com/Haivision/srt#commit=$_commit")
+sha256sums=('SKIP')
 
 pkgver() {
     cd $_basename
@@ -25,7 +23,7 @@ pkgver() {
 prepare() {
     cd $_basename
 
-    patch -Np1 -i ../no-rpath.diff
+    #patch -Np1 -i ../no-rpath.diff
 }
 
 build() {
@@ -33,12 +31,14 @@ build() {
     export CXX='g++ -m32'
     export PKG_CONFIG_PATH='/usr/lib32/pkgconfig'
 
-    cmake -Hsrt -Bbuild -GNinja \
+    cmake -Ssrt -Bbuild -GNinja \
         -DCMAKE_INSTALL_PREFIX=/usr \
         -DCMAKE_INSTALL_BINDIR=bin \
         -DCMAKE_INSTALL_LIBDIR=lib32 \
         -DCMAKE_INSTALL_INCLUDEDIR=include \
-        -DENABLE_TESTING=True
+        -DCMAKE_BUILD_TYPE=None \
+        -DENABLE_STATIC=ON \
+        -DENABLE_TESTING=ON
 
     cmake --build build
 }
@@ -50,7 +50,7 @@ check() {
 }
 
 package() {
-    DESTDIR="$pkgdir" cmake --build build --target install
+    DESTDIR="$pkgdir" cmake --install build
 
     cd "$pkgdir"/usr
 
