diff --git a/.SRCINFO b/.SRCINFO
index 9c8831e..b3a53fa 100644
--- a/.SRCINFO
+++ b/.SRCINFO
@@ -1,6 +1,6 @@
 pkgbase = lib32-openexr
 	pkgdesc = An high dynamic-range image file format library (32-bit)
-	pkgver = 2.5.1
+	pkgver = 2.5.2
 	pkgrel = 1
 	url = https://www.openexr.com/
 	arch = x86_64
@@ -13,8 +13,8 @@ pkgbase = lib32-openexr
 	depends = openexr
 	conflicts = lib32-ilmbase
 	replaces = lib32-ilmbase
-	source = openexr-2.5.1.tar.gz::https://github.com/openexr/openexr/archive/v2.5.1.tar.gz
-	sha256sums = 11f806bf256453e39fc33bd1cf1fa576a54f144cedcdd3e6935a177e5a89d02e
+	source = openexr-2.5.2.tar.gz::https://github.com/openexr/openexr/archive/v2.5.2.tar.gz
+	sha256sums = 5da8dff448d0c4a529e52c97daf238a461d01cd233944f75095668d6d7528761
 
 pkgname = lib32-openexr
 
diff --git a/PKGBUILD b/PKGBUILD
index 0b9cc5b..a29083a 100644
--- a/PKGBUILD
+++ b/PKGBUILD
@@ -3,7 +3,7 @@
 
 _basename=openexr
 pkgname=lib32-openexr
-pkgver=2.5.1
+pkgver=2.5.2
 pkgrel=1
 pkgdesc="An high dynamic-range image file format library (32-bit)"
 url="https://www.openexr.com/"
@@ -14,33 +14,26 @@ makedepends=('cmake' 'lib32-fltk' 'lib32-freeglut')
 conflicts=('lib32-ilmbase')
 replaces=('lib32-ilmbase')
 source=($_basename-$pkgver.tar.gz::"https://github.com/openexr/openexr/archive/v$pkgver.tar.gz")
-sha256sums=('11f806bf256453e39fc33bd1cf1fa576a54f144cedcdd3e6935a177e5a89d02e')
+sha256sums=('5da8dff448d0c4a529e52c97daf238a461d01cd233944f75095668d6d7528761')
 
 build() {
     export CC='gcc -m32'
     export CXX='g++ -m32'
     export PKG_CONFIG_PATH='/usr/lib32/pkgconfig'
 
-    mkdir -p build
-
-    cd build
-
-    cmake ../$_basename-$pkgver \
+    cmake -B build -S $_basename-$pkgver \
         -DCMAKE_INSTALL_PREFIX=/usr \
         -DCMAKE_INSTALL_LIBDIR=lib32 \
         -DPYILMBASE_ENABLE=false \
         -DOPENEXR_BUILD_UTILS=false
-
-    make
+    cmake --build build
 }
 
 package() {
-    cd build
-
-    make DESTDIR="${pkgdir}" install
+    DESTDIR="${pkgdir}" cmake --install build
 
     rm -r "${pkgdir}/usr/include"
     rm -r "${pkgdir}/usr/share"
 
-    install -D -m644 "$srcdir"/$_basename-$pkgver/LICENSE.md -t "${pkgdir}/usr/share/licenses/${pkgname}"
+    install -D -m644 $_basename-$pkgver/LICENSE.md -t "${pkgdir}/usr/share/licenses/${pkgname}"
 }
