diff --git a/.SRCINFO b/.SRCINFO
index 93ddfd9..bcf3b87 100644
--- a/.SRCINFO
+++ b/.SRCINFO
@@ -1,5 +1,5 @@
 pkgbase = lib32-bluez-libs
-	pkgver = 5.52
+	pkgver = 5.54
 	pkgrel = 1
 	url = http://www.bluez.org/
 	arch = x86_64
@@ -8,11 +8,11 @@ pkgbase = lib32-bluez-libs
 	makedepends = lib32-gcc-libs
 	makedepends = lib32-glib2
 	makedepends = lib32-systemd
-	source = https://www.kernel.org/pub/linux/bluetooth/bluez-5.52.tar.xz
-	source = https://www.kernel.org/pub/linux/bluetooth/bluez-5.52.tar.sign
+	source = https://www.kernel.org/pub/linux/bluetooth/bluez-5.54.tar.xz
+	source = https://www.kernel.org/pub/linux/bluetooth/bluez-5.54.tar.sign
 	source = refresh_adv_manager_for_non-LE_devices.diff
 	validpgpkeys = E932D120BC2AEC444E558F0106CA9F5D1DCF2659
-	sha512sums = 9024481910d1dee85578bf3d7a4fc4be2bece516e6216c91aa1a7ac18537f092b47c41c4a5506ddd7e279599d58002e98203c5439745c01df4dbc22a75234c84
+	sha512sums = e19d15d3a478a7af47c1921c8827843492e38787b1182152155bd3d8ad9e1d8ee25c5fda1f24e38c54ebbf946b09fe75007dca9a24d1c35f73303558e558dcbe
 	sha512sums = SKIP
 	sha512sums = c6ef673956963725edc52d667648e51df5b99f820e87705096b4b9338e8a4540722e734f1e8e67998c2fbbefec30645ff1fa064789c8a858f770d1214399561d
 
diff --git a/PKGBUILD b/PKGBUILD
index 58e8462..9f400e3 100644
--- a/PKGBUILD
+++ b/PKGBUILD
@@ -6,23 +6,21 @@
 
 pkgbase=lib32-bluez-libs
 pkgname=("${pkgbase}" 'lib32-bluez-plugins')
-pkgver=5.52
+pkgver=5.55
 pkgrel=1
 url="http://www.bluez.org/"
 arch=('x86_64')
 license=('LGPL2.1')
 makedepends=('lib32-dbus' 'lib32-gcc-libs' 'lib32-glib2' 'lib32-systemd')
-source=("https://www.kernel.org/pub/linux/bluetooth/bluez-${pkgver}.tar."{xz,sign}
-        "refresh_adv_manager_for_non-LE_devices.diff")
+source=("https://www.kernel.org/pub/linux/bluetooth/bluez-${pkgver}.tar."{xz,sign})
 # see https://www.kernel.org/pub/linux/bluetooth/sha256sums.asc
-sha512sums=('9024481910d1dee85578bf3d7a4fc4be2bece516e6216c91aa1a7ac18537f092b47c41c4a5506ddd7e279599d58002e98203c5439745c01df4dbc22a75234c84'
-            'SKIP'
-            'c6ef673956963725edc52d667648e51df5b99f820e87705096b4b9338e8a4540722e734f1e8e67998c2fbbefec30645ff1fa064789c8a858f770d1214399561d')
+sha512sums=('9423cb60d15a6f068838497a1eaea9f5a32d70c07191c313ba821a6919d6e0c436ada4f547cc5f2db5eacc0123429ad54851f57df2554f61fa293743ec14a033'
+            'SKIP')
 validpgpkeys=('E932D120BC2AEC444E558F0106CA9F5D1DCF2659') # Marcel Holtmann <marcel@holtmann.org>
 
 prepare() {
   cd "bluez-${pkgver}"
-  patch -Np1 -i ../refresh_adv_manager_for_non-LE_devices.diff
+  #patch -Np1 -i ../refresh_adv_manager_for_non-LE_devices.diff
 }
 
 build() {
