diff --git a/.SRCINFO b/.SRCINFO
index 455fa31..1da59a0 100644
--- a/.SRCINFO
+++ b/.SRCINFO
@@ -1,6 +1,6 @@
 pkgbase = lib32-openjpeg2
-	pkgdesc = An open source JPEG 2000 codec, version 2.3.1
-	pkgver = 2.3.1
+	pkgdesc = An open source JPEG 2000 codec, version 2.4.0
+	pkgver = 2.4.0
 	pkgrel = 1
 	url = http://www.openjpeg.org
 	arch = x86_64
@@ -12,8 +12,8 @@ pkgbase = lib32-openjpeg2
 	makedepends = lib32-lcms2
 	depends = openjpeg2
 	depends = lib32-zlib
-	source = https://github.com/uclouvain/openjpeg/archive/v2.3.1.tar.gz
-	md5sums = 3b9941dc7a52f0376694adb15a72903f
+	source = https://github.com/uclouvain/openjpeg/archive/v2.4.0.tar.gz
+	b2sums = ab8907638ac041ce7dcbcbcd9624ea5e4b7542c9ec38a850e363c071c27c4bc8b16e207700b12e67d8d32bdd9b0838735bede27084090ce95105d32c539b09cf
 
 pkgname = lib32-openjpeg2
 
diff --git a/PKGBUILD b/PKGBUILD
index 5665db0..95665a9 100644
--- a/PKGBUILD
+++ b/PKGBUILD
@@ -6,7 +6,7 @@ _setFullLibdir="${_setPrefix}/${_setLibdir}"
 _pkgbase=openjpeg2
 
 pkgname=${_setLibdir}-${_pkgbase}
-pkgver=2.3.1
+pkgver=2.4.0
 pkgrel=1
 pkgdesc="An open source JPEG 2000 codec, version ${pkgver}"
 arch=('x86_64')
@@ -15,7 +15,7 @@ url="http://www.openjpeg.org"
 makedepends=('cmake' 'lib32-gcc-libs' 'lib32-libpng'  'lib32-libtiff' 'lib32-lcms2')
 depends=("${_pkgbase}" 'lib32-zlib')
 source=("https://github.com/uclouvain/openjpeg/archive/v${pkgver}.tar.gz")
-md5sums=("3b9941dc7a52f0376694adb15a72903f")
+b2sums=('ab8907638ac041ce7dcbcbcd9624ea5e4b7542c9ec38a850e363c071c27c4bc8b16e207700b12e67d8d32bdd9b0838735bede27084090ce95105d32c539b09cf')
 
 build() {
   export CFLAGS="-m32"
