diff --git a/.SRCINFO b/.SRCINFO
index c787f19..f753755 100644
--- a/.SRCINFO
+++ b/.SRCINFO
@@ -1,6 +1,6 @@
 pkgbase = lib32-zbar
 	pkgdesc = Application and library for reading bar codes from various sources. 32bit libraries only version
-	pkgver = 0.23
+	pkgver = 0.23.1
 	pkgrel = 1
 	url = https://github.com/procxx/zbar
 	arch = x86_64
@@ -13,8 +13,8 @@ pkgbase = lib32-zbar
 	depends = lib32-dbus
 	depends = lib32-libsm
 	depends = lib32-libxv
-	source = zbar-0.23.tar.gz::https://github.com/mchehab/zbar/archive/0.23.tar.gz
-	sha512sums = bc2667a2a5a5f2e9fd3f39e57bf1d7f860b20a0b5faffc38791d222a6ff0414aff880295ce5ccb92acce7250503f2a7f48ac12651cf82cfb1933b302da26e393
+	source = zbar-0.23.1.tar.gz::https://github.com/mchehab/zbar/archive/0.23.1.tar.gz
+	sha512sums = ae7741cf750a10cf53dc11abcd482c3885507153ee37f6e3364ed5ed72184ebb009560b8c40d8090603a551fb681700a962838a59ce77d005d080ee49fbfa54b
 
 pkgname = lib32-zbar
 
diff --git a/PKGBUILD b/PKGBUILD
index 2669354..49e4e97 100644
--- a/PKGBUILD
+++ b/PKGBUILD
@@ -6,7 +6,7 @@
 # Mantainer: Lorenzo Ferrillo <lorenzofer at live dot it>
 _pkgname=zbar
 pkgname=lib32-zbar
-pkgver=0.23
+pkgver=0.23.1
 pkgrel=1
 pkgdesc="Application and library for reading bar codes from various sources. 32bit libraries only version"
 arch=('x86_64')
@@ -16,7 +16,7 @@ depends=('zbar' 'lib32-imagemagick' 'lib32-v4l-utils' 'lib32-dbus' 'lib32-libsm'
 makedepends=('docbook-xsl' 'xmlto')
 source=("$_pkgname-$pkgver.tar.gz::https://github.com/mchehab/zbar/archive/$pkgver.tar.gz"
 )
-sha512sums=('bc2667a2a5a5f2e9fd3f39e57bf1d7f860b20a0b5faffc38791d222a6ff0414aff880295ce5ccb92acce7250503f2a7f48ac12651cf82cfb1933b302da26e393'
+sha512sums=('ae7741cf750a10cf53dc11abcd482c3885507153ee37f6e3364ed5ed72184ebb009560b8c40d8090603a551fb681700a962838a59ce77d005d080ee49fbfa54b'
 )
 prepare() {
   cd zbar-$pkgver
