diff --git a/tgcalls/group/StreamingPart.cpp b/tgcalls/group/StreamingPart.cpp
index a7f2e00..e3710ca 100644
--- a/tgcalls/group/StreamingPart.cpp
+++ b/tgcalls/group/StreamingPart.cpp
@@ -1,5 +1,3 @@
-#include "StreamingPart.h"
-
 #include "rtc_base/logging.h"
 #include "rtc_base/third_party/base64/base64.h"
 
@@ -14,6 +12,8 @@ extern "C" {
 #include <map>
 #include <stdint.h>
 
+#include "StreamingPart.h"
+
 namespace tgcalls {
 
 namespace {
